### ISSUE TITLE
@NavTypeSerializer not used from separate module

### ISSUE THREAD

**Issue by kassim** [raamcosta/compose-destinations#173]
I've created it serializer 
`@NavTypeSerializer class BitmapTypeSerializer : DestinationsNavTypeSerializer<Bitmap>`
(I know its probably inefficient to serialise a bitmap between screens but that's a different point)

I also have my composable destination
```
@Composable
@Destination
fun ReviewScreen(destinationsNavigator: DestinationsNavigator, bitmap: Bitmap?) {
```

when I put `BitmapTypeSerializer` next to `ReviewScreen` in the same package the serialization works fine, but when I move the Serializer to a `navigation` module that my feature module depends on, the codegen decides to use `DefaultParcelableNavTypeSerializer` instead of `BitmapTypeSerializer`

**Comment by raamcosta** [comment]
Hey 👋 

Yeah at the moment this is expected. Ksp tasks are completely isolated inside each module. So it only knows about classes of that module. 

But btw passing bitmap really is not something you want to do, the Android Binder only has  500 kb of space, so you can easily get exception if you’re using it for something like a bitmap.

Anyway I’ll keep this open so I can investigate if there is a way to know about classes of other modules on ksp task.

**Comment by raamcosta** [comment]
I was developing a feature for this, but I realized it's stupid to have a specific API just for this.
Because ideally, we register a nav type serializer globally for that type in that module and for that, there is nothing simpler then what is already possible.

```kotlin
@NavTypeSerializer
object ClassFromOtherModuleSerializer : DestinationsNavTypeSerializer<ClassFromOtherModule> by NavTypeSerializerDeclaredOnTheOtherModule
```

**Comment by raamcosta** [comment]
@kassim @Nek-12 What do you guys think? Am I missing something?

**Comment by Nek-12** [comment]
Yeah I think what I consider to be a "workaround" looks good from the code standpoint, but still, this does not solve the problem of some people wanting to use different serializers for different composables or even parameters. I believe such a use case is going to appear at some point.  Kotlinx.serialization doesn't have anything like the module serializer, but it does have a file-wide, class level and parameter-level annotations, which is I think the most flexibility there possibly could be.

**Comment by pawelsmagala** [comment]
@raamcosta bump on that - would be nice to be able to define global serializer for specific type that is reused in every module
